cpp/.clang-tidy

```diff
@@ -0,0 +1,31 @@
+Checks: >
```

Comment:
Minor suggestion: We can enable this in Github action to do some code style pre-check.

Reply:
Sure. I'll open another PR for that.

Reply:
Issue: https://github.com/oap-project/gluten/issues/1599